app/cdap/components/ConnectionList/__tests__/ConnectionList.test.tsx

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import { render, screen } from '@testing-library/react';
+import * as apiHelpersForExploreConnection from 'components/Connections/Browser/GenericBrowser/apiHelpers';
+import * as apiHelpers from 'components/Connections/Browser/SidePanel/apiHelpers';
+import * as reducer from 'components/Connections/Create/reducer';
+import history from 'app/cdap/services/history';
+import React from 'react';
+import { Route, Router, Switch } from 'react-router';
+import ConnectionList from '..';
+import {
+  connectionListDummyResFile,
+  connectionListDummyResPostGresSql,
+  mockDataForExploreConnection,
+  mockResponseForFetchConnectors,
+} from 'components/ConnectionList/mock/mockDataForConnectionList';
+
+describe('It should test Connection List component', () => {
+  beforeEach(() => {
+    const dummyRes = new Map();
+    dummyRes.set('PostgreSql', connectionListDummyResPostGresSql);
+    dummyRes.set('File', connectionListDummyResFile);
+    jest.spyOn(apiHelpers, 'getCategorizedConnections').mockReturnValue(Promise.resolve(dummyRes));
+    jest.spyOn(apiHelpersForExploreConnection, 'exploreConnection').mockImplementation(() => {
+      return Promise.resolve(mockDataForExploreConnection);
+    });
+  });
+  it('Should render Connection List component', () => {
+    const container = render(
+      <Router history={history}>
+        <Switch>
+          <Route>
+            <ConnectionList />
+          </Route>
+        </Switch>
+      </Router>
+    );
+    expect(container).toBeDefined();
+  });
+  it('Should render Connection List component', () => {
+    const dummyRes = new Map();
+    dummyRes.set('PostgreSql', connectionListDummyResPostGresSql);
+    dummyRes.set('File', connectionListDummyResFile);
+    jest
+      .spyOn(reducer, 'fetchConnectors')
+      .mockReturnValue(Promise.resolve(mockResponseForFetchConnectors));
+
+    jest.spyOn(apiHelpers, 'getCategorizedConnections').mockReturnValue(Promise.resolve(dummyRes));
+
+    const container = render(
+      <Router history={history}>
+        <Switch>
+          <Route>
+            <ConnectionList />
+          </Route>
+        </Switch>
+      </Router>
+    );
```

Comment:
can we add more test cases plz to cover all cases 
TO SECURE THE CODE

not just for coverage report